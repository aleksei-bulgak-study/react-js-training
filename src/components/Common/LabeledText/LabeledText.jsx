import React from 'react';
import PropTypes from 'prop-types';
import './LabeledText.css';

const LabeledText = ({ id, title, value, className }) => (
  <label htmlFor={id} className='labeled-title'>
    {title}
    <p id={id} className={className}>
      {value}
    </p>
  </label>
);

LabeledText.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string,
};

LabeledText.defaultProps = {
  value: '',
  className: 'labeled-text',
};

export default LabeledText;
