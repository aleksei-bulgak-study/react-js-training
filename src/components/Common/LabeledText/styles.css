.labeled-text {
  color: var(--white);
}
