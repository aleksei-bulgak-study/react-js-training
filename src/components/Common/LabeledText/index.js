import LabeledText from './LabeledText';
export default LabeledText;
