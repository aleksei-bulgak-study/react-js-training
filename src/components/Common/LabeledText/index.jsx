import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const LabeledText = ({ id, title, value }) => (
  <label htmlFor={id}>
    {title}
    <p id={id}>{value}</p>
  </label>
);

LabeledText.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
};

LabeledText.defaultProps = {
  value: '',
};

export default LabeledText;
