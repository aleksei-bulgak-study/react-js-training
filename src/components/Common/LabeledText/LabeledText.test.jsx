import React from 'react';
import { shallow } from 'enzyme';
import LabeledText from './LabeledText';

describe('LabeledText', () => {
  test('when all props passed then all props rendered', () => {
    const textComponent = shallow(
      <LabeledText
        id="testId"
        title="testTitle"
        value="testValue"
        className="additionalClass"
      />,
    );
    expect(textComponent.find('label.labeled-title').text()).toContain('testTitle');
    expect(textComponent.find('label.labeled-title > p#testId').hasClass('additionalClass')).toBeTruthy();
    expect(textComponent.find('label.labeled-title > p#testId').text()).toEqual('testValue');
    expect(textComponent.html()).toMatchSnapshot();
  });
});
