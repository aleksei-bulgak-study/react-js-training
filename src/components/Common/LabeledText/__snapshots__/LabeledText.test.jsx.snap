// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledText when all props passed then all props rendered 1`] = `"<label for=\\"testId\\" class=\\"labeled-title\\">testTitle<p id=\\"testId\\" class=\\"additionalClass\\">testValue</p></label>"`;
